cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/JepSingleton.kt

```diff
@@ -112,47 +120,42 @@ object JepSingleton {
-                    ),
-                    File(
-                        "${System.getProperty("user.home")}/.virtualenvs/${virtualEnv}/lib/python3.10/site-packages/jep/libjep.so"
-                    ),
-                    File(
-                        "${System.getProperty("user.home")}/.virtualenvs/${virtualEnv}/lib/python3.10/site-packages/jep/libjep.jnilib"
-                    ),
-                    File(
-                        "${System.getProperty("user.home")}/.virtualenvs/${virtualEnv}/lib/python3.9/site-packages/jep/libjep.so"
-                    ),
-                    File(
-                        "${System.getProperty("user.home")}/.virtualenvs/${virtualEnv}/lib/python3.9/site-packages/jep/libjep.jnilib"
-                    ),
-                    File("/usr/lib/libjep.so"),
-                    File("/Library/Java/Extensions/libjep.jnilib")
-                )
-
-            wellKnownPaths.forEach {
-                if (it.exists()) {
-                    // Jep's configuration must be set before the first instance is created. Later
-                    // calls
-                    // to setJepLibraryPath and co result in failures.
-                    MainInterpreter.setJepLibraryPath(it.path)
-                    config.addIncludePaths(
-                        it.toPath().parent.parent.toString()
-                    ) // this assumes that the python code is also at the library's location
+            val knownPathsToJep =
+                listOf("3.11", "3.10", "3.9").flatMap {
+                    listOf(
+                        libJepSoFromVirtualEnv(it),
+                        libJepJniFromVirtualEnv(it),
+                        macOsInstallationPath(it)
+                    )
+                } + otherPathsToJepLibrary()
+
+            knownPathsToJep
+                .map { File(it) }
+                .forEach {
+                    if (it.exists()) {
+                        // Jep's configuration must be set before the first instance is created.
+                        // Later calls to setJepLibraryPath and co result in failures.
+                        MainInterpreter.setJepLibraryPath(it.path)
+                        config.addIncludePaths(
+                            it.toPath().parent.parent.toString()
+                        ) // this assumes that the python code is also at the library's location
+                    }
                 }
-            }
         }
     }
+
+    private fun libJepSoFromVirtualEnv(version: String) =
+        "${System.getProperty("user.home")}/.virtualenvs/$virtualEnv/lib/python${version}/site-packages/jep/$libJepSo"
+
+    private fun libJepJniFromVirtualEnv(version: String) =
+        "${System.getProperty("user.home")}/.virtualenvs/$virtualEnv/lib/python${version}/site-packages/jep/$libJepJniLib"
+
+    private fun macOsInstallationPath(version: String) =
+        "/opt/homebrew/lib/python${version}/site-packages/jep/$libJepJniLib"
```

Comment:
without this path - there will be a problem to run tests locally on latest Mac with brew installations of python and jep